rfcs/groups/groups.mmd

```diff
@@ -0,0 +1,102 @@
+sequenceDiagram
+  participant M as Existing<br>member (M)
+  participant MA as Existing<br>member<br>agent (MA)
+  participant A as Alice (A)
+  participant AA as Alice's<br>agent (AA)
+  participant BA as Bob's<br>agent (BA)
+  participant B as Bob (B)
+
+  note over A, AA: 1. create new group (no members)
+  A ->> AA: gidA? GNEW gInfo<br>(gidA - conn alias of this group for A,<br>can be generated by the agent.<br>Does it share namespace with connections?)
+  AA ->> A: gidA OK<br>(or GOK?)
+
+  note over A, BA: 2. add Bob to group
+  A ->> AA: gidA GADD idAB<br>(idAB - conn alias A has for B)
+  
+  note over AA: initiate internal connection gidAB for B in group
+
+  AA ->> BA: via idAB: G_INV gidABinv gMeta gInfo<br>(gMeta contains number of members - TBC what else<br>should not contain date of creation or who created it)
+  BA ->> B: idBA GINV invID gMeta gInfo
+  B ->> BA: gidB? GJOIN invID<br>(could be also GREJECT invID?)
+  BA ->> B: gidB OK
+
+  BA ->> AA: establish internal connection gidBA (using gidABinv) for A in group<br>(intermal means that it is not visible to the clients. Separate namespace?)
+ 
+  AA -> A: gidA GCON idAB
+
+  note over M, B: below steps happen for each existing member M
+
+  AA ->> BA: via gidBA: GM_ADD lvlAM midABM|_<br>(midABM - how A identifies member M to B,<br>lvlAM - connection level, 0 - direct, 1 - via smbdy etc.)
+  
+  note over BA: initiate connection idBM for direct messages and<br> internal gidBM for group messages between B and M
+  
+  BA ->> AA: via gidBA: GM_INV midABM|BAM idBMinv/gidBMinv<br>(midBAM - how B identifies member M to A<br>idBMinv/gidBMinv - invitations to connect for member M)
```

Comment:
`midABM` -> `midBAM`

Reply:
idBMinv/gidBMinv - swap B and M? this is invitation for connection from M to B - wouldn't it be better to keep these indices consistent in sender to receiver direction, or I'm missing something?

Reply:
Discussed - we will use a single unique ID per member of the group, to allow member mentions in group messages.